qiskit/pulse/schedule.py

```diff
@@ -141,17 +141,42 @@ def _instructions(self, time: int = 0) -> Iterable[Tuple[int, 'Instruction']]:
         for insert_time, child_sched in self._children:
             yield from child_sched._instructions(time + insert_time)
 
-    def union(self, *schedules: List[ScheduleComponent],
+    def union(self, *schedules: Union[ScheduleComponent, Tuple[int, ScheduleComponent]],
               name: Optional[str] = None) -> 'Schedule':
-        """Return a new schedule which is the union of `self` and `schedule`.
+        """Return a new schedule which is the union of children of both `self` and `schedules`.
 
         Args:
             *schedules: Schedules to be take the union with this `Schedule`.
             name: Name of the new schedule. Defaults to name of self
         """
         if name is None:
             name = self.name
-        return Schedule(self, *schedules, name=name)
+        new_sched = Schedule(name=name)
+        new_sched._union((0, self))
+        for sched_pair in schedules:
+            if not isinstance(sched_pair, tuple):
+                sched_pair = (0, sched_pair)
+            new_sched._union(sched_pair)
+        return new_sched
+
+    def _union(self, other: Tuple[int, ScheduleComponent]) -> 'Schedule':
+        """Mutably union `self` and `other` Schedule with shift time.
+
+        Args:
+            other: Schedule with shift time to be take the union with this `Schedule`.
+        """
+        shift_time, sched = other
+        if isinstance(sched, Schedule):
+            shifted_children = sched._children
+            if shift_time != 0:
+                shifted_children = tuple((t + shift_time, child) for t, child in shifted_children)
+            self.__children += shifted_children
+        else:  # isinstance(sched, Instruction)
+            self.__children += (other,)
```

Comment:
I think this should be `+= (shift_time, other)`, right?

Reply:
I think I'm not wrong in line 175. The rhs should have the same type as `self.__children: Tuple[Tuple[int, ScheduleComponent]]`.

Reply:
Ok, got it :)